tensorflow/compiler/mlir/tosa/transforms/convert_tfl_uint8.cc

```diff
@@ -130,6 +130,63 @@ struct ConvertUint8QConstOp : public RewritePattern {
   }
 };
 
+namespace {
+
+// returns true iff @a type is a shaped type with element type that is uint8
+// if it is, then return the rescaled type, uint8_zp, and output_zp to use to
+// rescale type to signed type with adjusted zero point.
+bool IsShapedUint8Type(OpBuilder& builder, const Type type, Type& rescaled_type,
+                       int32_t& uint8_zp, int32_t& output_zp) {
+  auto uint8_type = type.dyn_cast<mlir::ShapedType>();
+  if (!uint8_type) return false;
+
+  auto element_type = uint8_type.getElementType();
+  auto uint8_element_quant_type =
+      element_type.dyn_cast<mlir::quant::UniformQuantizedType>();
+  bool is_uint8_element_quant_type =
+      uint8_element_quant_type && !uint8_element_quant_type.isSigned() &&
+      uint8_element_quant_type.getStorageTypeIntegralWidth() == 8;
+  bool is_uint8_element_type = element_type.isUnsignedInteger(8);
+  if (!is_uint8_element_quant_type && !is_uint8_element_type) return false;
+
+  // type has uint8 element type
+  if (is_uint8_element_quant_type) {
+    double type_range_min =
+        static_cast<double>(uint8_element_quant_type.getStorageTypeMin() -
+                            uint8_element_quant_type.getZeroPoint()) *
+        uint8_element_quant_type.getScale();
+    double type_range_max =
+        static_cast<double>(uint8_element_quant_type.getStorageTypeMax() -
+                            uint8_element_quant_type.getZeroPoint()) *
+        uint8_element_quant_type.getScale();
+    bool narrow_range =
+        uint8_element_quant_type.getStorageTypeMin() == 1 ? true : false;
+
+    rescaled_type = uint8_type.clone(buildQTypeFromMinMax(
+        builder, uint8_element_quant_type.getExpressedType(),
+        builder.getF64FloatAttr(type_range_min),
+        builder.getF64FloatAttr(type_range_max),
+        builder.getI32IntegerAttr(
+            uint8_element_quant_type.getStorageTypeIntegralWidth()),
+        0, true /* signed */, builder.getBoolAttr(narrow_range)));
```

Comment:
This should include /*arg=*/ for any non obvious arg. Specifically the `0` and `true`  should both have it.

Reply:
fixed